include/grpc++/client_context.h

```diff
@@ -107,6 +108,8 @@ class ClientContext {
     creds_ = creds;
   }
 
+  std::unique_ptr<const AuthContext> auth_context() const;
```

Comment:
Is there any advantage of using a `unique_ptr` as opposed to just:
`const AuthContext* auth_context() const;`
?